modules/benchmark/R/clone_run.R

```diff
@@ -0,0 +1,115 @@
+  inputs_all <- tbl(bety, "modeltypes_formats") %>% filter(model.new$modeltype_id == modeltype_id) %>% select(one_of("tag", "required")) %>% collect
+  inputs_req <- inputs_all$tag[inputs_all$required]
+  
+  #------ Update model ------#
+  
+  settings.new$model$id <- model.new$id
+  
+  #------ Update inputs ------#
+  
+  for(tag in names(settings.old$run$inputs)){
+    
+    # Check that the given input is required for the new model
+    if(tag %in% inputs_all$tag){
+      
+      # Check lookup table to see what format input needs to be in 
+      # For example for met we need the CF standard 
+      # For IC it depends on pool vs cohort - may need special cases
+      tags_std <- read.csv(
+        system.file("clone_run_inputs_lookup.csv", package = "PEcAn.benchmark"), 
+        stringsAsFactors = FALSE) 
+      new_format <- c()
+      
+      # Check if old model input is in the correct format
+      old.input <- tbl(bety, "inputs") %>% 
+        filter(settings.old$run$inputs[[tag]]$id == id) %>% 
+        select( one_of("id", "format_id", "parent_id")) %>% collect()
+      
+      if(!is.na(old.input$parent_id)){
+        parent.input <- tbl(bety, "inputs") %>% 
+          filter(old.input$parent_id == id) %>% 
+          select( one_of("id", "format_id", "parent_id")) %>% collect()
+      }else{
+        parent.input <- list(format_id = NA)
+      }
+      
+      if(old.input$format_id == new_format){
+        # The input id points to a file that is already in the right format to be converted
+        settings.new$run$inputs[[tag]]$id  <- old.input$id 
+      }else if(parent.input$format_id == new_format){
+        # The parent id points to a file that is in the right format to be converted
+        settings.new$run$inputs[[tag]]$id  <- parent.input$id 
+      }else if(is.na(old.input$parent_id)){
+        # If the input has no parent id, assume it's raw data (is that safe???)
+        settings.new$run$inputs[[tag]]$id  <- old.input$id
+      }else{
+        # This shouldn't ever happen
+        # I assume model specific input shoud always have the standard as the parent
+      }
+      
+    }else{
+      # If input not needed, the tag could still be added to the new settings 
+      # and be commented out inthe XML, but that would be difficult
+      # to implement with a list object. I propose moivng it to an "omitted" list 
+      # that could be written out in an XML or in some sort of report
+    }
+  }
+  
+  # Check that all required inputs are in settings
+  missing_inputs <- setdiff(names(settings.new$run$inputs), inputs_req)
+  if(length(missing_inputs)>0){
+    PEcAn.utils::logger.warn("Required inputs were not specified in the old run settings")
+    # If there are obvious defaults, use them to fill in missing inputs
+    # Make sure the XML can't be run if there are missing inputs
+  }
+  
+  #------ Update PFTs ------#
+
+  old.pfts <- unlist(lapply(seq_along(pfts), function(x) settings$pfts[[x]][["name"]]))
```

Comment:
I think we want id, not name. If only name is present we probably have to look up ID since name is not unique (two different models can have PFTs with the same name)